ngraph_bridge/ngraph_encapsulate_impl.cc

```diff
@@ -0,0 +1,445 @@
+        output_tensors_bytes_free += next_output.second->get_size_in_bytes();
+        next_output.second.reset();
+      }
+      m_ng_exec_output_cache_map.erase(evicted_ng_exec);
+      m_lru.pop_back();
+      NGRAPH_VLOG(1) << "NGRAPH_TF_MEM_PROFILE:  OP_ID: " << my_instance_id
+                     << " Cluster: " << m_name << " Input Tensors freed: "
+                     << input_tensors_bytes_free / (1024 * 1024) << " MB"
+                     << " Output Tensors freed: "
+                     << output_tensors_bytes_free / (1024 * 1024) << " MB";
+    }  // cache eviction if cache size greater than cache depth
+
+    BackendManager::LockBackend(m_op_backend_name);
+
+    ngraph::Event event_compile("Compile nGraph", m_name, "");
+    try {
+      ng_exec = op_backend->compile(ng_function);
+    } catch (const std::exception& exp) {
+      BackendManager::UnlockBackend(m_op_backend_name);
+      NgraphSerialize("tf_function_error_" + m_name + ".json", ng_function);
+      return errors::Internal("Caught exception while compiling op_backend: ",
+                              exp.what(), "\n");
+    } catch (...) {
+      BackendManager::UnlockBackend(m_op_backend_name);
+      NgraphSerialize("tf_function_error_" + m_name + ".json", ng_function);
+      return errors::Internal("Error in compiling op_backend\n");
+    }
+    BackendManager::UnlockBackend(m_op_backend_name);
+    event_compile.Stop();
+    ngraph::Event::write_trace(event_compile);
+
+    SetNgExecMap(signature, ng_exec);
+    // caching ng_function to serialize to ngraph if needed
+    SetNgFunctionMap(ng_exec, ng_function);
+
+    m_lru.push_front(signature);
+    // Memory after
+    MemoryProfile(vm, rss);
+    auto delta_vm_mem = vm - vm0;
+    auto delta_res_mem = rss - rss0;
+    NGRAPH_VLOG(1) << "NGRAPH_TF_CACHE_PROFILE: OP_ID: " << my_instance_id
+                   << " Cache length: " << m_ng_exec_map.size()
+                   << "  Cluster: " << m_name << " Delta VM: " << delta_vm_mem
+                   << "  Delta RSS: " << delta_res_mem
+                   << "  Function size: " << function_size
+                   << " KB Total RSS: " << rss / (1024 * 1024) << " GB "
+                   << " VM: " << vm / (1024 * 1024) << " GB" << endl;
+  }  // end of input signature not found in m_ng_exec_map
+  else {
+    // Found the input signature in m_ng_exec_map, use the cached executable
+    // Update the m_lru
+    if (signature != m_lru.front()) {
+      m_lru.remove(signature);
+      m_lru.push_front(signature);
+    }
+    ng_exec = it->second;
+  }
+  return Status::OK();
+}
+
+// Allocate tensors for input arguments. Creates ngraph input tensors using
+// tensorflow tensors required to execute ngraph function
+Status NGraphEncapsulateImpl::AllocateNGInputTensors(
+    const std::vector<Tensor>& tf_input_tensors,
+    const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+    ng::runtime::Backend* op_backend,
```

Comment:
`ng::runtime::Backend* op_backend` -> `const ng::runtime::Backend* const op_backend`

The pointer value should not change, and the pointer should not modify its contents. or whatever compiles

I use the first const to understand what things are inputs and what are outputs of these functions. Hence would prefer a const